src/server.ts

```diff
@@ -287,18 +287,46 @@ export class Server {
    *
    * @param {string} hash - The hex-encoded hash of the transaction to check.
    *
-   * @returns {Promise<SorobanRpc.GetTransactionResponse>} Returns a
-   *    promise to the {@link SorobanRpc.GetTransactionResponse} object
-   *    with the status, result, and other details about the transaction.
+   * @returns {Promise<SorobanRpc.GetTransactionResponse>} Returns a promise to
+   *    the {@link SorobanRpc.GetTransactionResponse} object with the status,
+   *    result, and other details about the transaction. Raw XDR fields are
+   *    parsed into their appropriate structures wherever possible.
    */
   public async getTransaction(
     hash: string,
   ): Promise<SorobanRpc.GetTransactionResponse> {
-    return await jsonrpc.post(
+    return await jsonrpc.post<SorobanRpc.RawGetTransactionResponse>(
```

Comment:
Nit: This fn would be much easier to read if you used async/await instead of mixing it with `.then` continuations...

```ts
const raw = await jsonrpc.post<SorobanRpc.RawGetTransactionResponse>(/* ... */);
let successInfo: any = {};
if (raw.status === 'SUCCESS') {
  // ...
}

// ...
```

Reply:
nice, fixed up!